### PR TITLE
[FLINK-29126][hive] Fix wrong logic for spliting file for orc format

### PR DESCRIPTION
**Pull request by luoyuxia**
## What is the purpose of the change
To fix the file split optimizatioin doesn't work for orc format.

## Brief change log
  -  Change`"orc".equalsIgnoreCase(serializationLib)`  to `!serializationLib.toLowerCase().contains("orc")` to check whether it's orc format or not, since the serializationLib for orc format is actually `OrcSerde` .

  -  optimization that enables multiple threads to calcalute total files' size .

## Verifying this change
UT in [HiveSourceFileEnumeratorTest.java](https://github.com/apache/flink/pull/20694/files#diff-059eba29b7890b2ab4886612e37be7ca35f255ce8519a5afb43d1a3a9f34b401)

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: docs/content/docs/connectors/table/hive/hive_read_write.md [hunk 1]
```diff
@@ -198,8 +198,13 @@ Users can do some performance tuning by tuning the split's size with the follow
     </tr>
   </tbody>
 </table>
-
-**NOTE**: Currently, these two configurations only works for the Hive table stored as ORC format.
+{{< hint warning >}}
+**NOTE**:
+- To tune the split's size, Flink will first get all files' size for all partitions.
+  If there are too many partitions, it maybe time-consuming,
+  then you can configure the job configuration `table.exec.calculate-partition-size.thread-num` (3 by default) to a bigger value to enable more threads to speed the process.
```
Comment (godfreyhe) [inline docs/content/docs/connectors/table/hive/hive_read_write.md:205]: speed up

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveOptions.java [hunk 2]
```diff
@@ -96,6 +96,12 @@ public class HiveOptions {
                                     + " When the value is over estimated, Flink will tend to pack Hive's data into less splits, which will be helpful when Hive's table contains many small files."
                                     + " And vice versa. It only works for the Hive table stored as ORC format.");
 
+    public static final ConfigOption<Integer> TABLE_EXEC_HIVE_CALCULATE_PARTITION_SIZE_THREAD_NUM =
+            key("table.exec.calculate-partition-size.thread-num")
```
Comment (godfreyhe) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveOptions.java:100]: table.exec.hive.calculate-partition-size.thread-num

**Review by godfreyhe — COMMENTED** [review]
Thanks for the fix, I left some comments

**Comment by luoyuxia** [comment]
@godfreyhe I have addressed your comments. Could you please help review again when you're free.

**Review by godfreyhe — APPROVED** [review]
Thanks for the update, LGTM

**Pull request closed without merging**